ast/src/parsed/folder.rs

```diff
@@ -0,0 +1,45 @@
+use super::asm::{
+    ASMModule, ASMProgram, Import, Machine, Module, ModuleStatement, SymbolDefinition, SymbolValue,
+};
+
+pub trait Folder<T> {
```

Comment:
Isn't this more like a `try_map`? For fold I would assume that you accumulate a list into a single item using a function that takes two arguments.

Anyway, could you please document a little?

Reply:
I'm wondering if a mutable visitor would not be better here.

Reply:
Ok, it's not a unified type like for expression, so a visitor would not make too much sense.